Sources/PopupBridge/NetworkClient.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+protocol Networkable {
+    func post<T: Encodable>(url: URL, body: T) async throws
+}
+
+final class NetworkClient: Networkable {
+    
+    // MARK: - Private Properties
+    
+    private let session: Sessionable
+    
+    // MARK: - Initializer
+    
+    init(session: Sessionable = URLSession.shared) {
+        self.session = session
+    }
+    
+    // MARK: - Internal Methods
+    
+    func post<T: Encodable>(url: URL, body: T) async throws {
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        request.allHTTPHeaderFields = ["Content-Type": "application/json"]
+        
+        do {
+            let encodedBody = try JSONEncoder().encode(body)
+            request.httpBody = encodedBody
+        } catch let encodingError {
+            throw NetworkError.encodingError(encodingError)
+        }
```

Comment:
I think we can use the shorthand here that swift gives us for free:
```suggestion
        } catch {
            throw NetworkError.encodingError(error)
        }
```

Reply:
Updated: [9a5e980](https://github.com/braintree/popup-bridge-ios/pull/75/commits/9a5e980c1a902a7c03f32083c25e6f3428cc4b30) since we have a strongly typed, this catch is unnecessary